compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -73,17 +73,18 @@ impl<'a> Interpreter<'a> {
             });
         }
 
-        if meta.kind != FunctionKind::Normal {
-            return self.call_builtin(function, arguments, location);
-        }
-
-        let parameters = meta.parameters.0.clone();
-        for ((parameter, typ, _), (argument, arg_location)) in parameters.iter().zip(arguments) {
-            self.define_pattern(parameter, typ, argument, arg_location)?;
-        }
+        let result = if meta.kind != FunctionKind::Normal {
+            self.call_builtin(function, arguments, location)?
+        } else {
+            let parameters = meta.parameters.0.clone();
+            for ((parameter, typ, _), (argument, arg_location)) in parameters.iter().zip(arguments)
+            {
+                self.define_pattern(parameter, typ, argument, arg_location)?;
+            }
```

Comment:
I think it'd be simpler if we moved the `enter_function` call after the call_builtin line instead